concent_api/verifier/decorators.py

```diff
@@ -24,3 +28,38 @@ def wrapper(*args, **kwargs):
             # Remove any files left in VERIFIER_STORAGE_PATH.
             clean_directory(settings.VERIFIER_STORAGE_PATH)
     return wrapper
+
+
+def ensure_retry_of_locked_calls(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        repeat_handler = RepeatHandler()
+        wait_time_in_sec = 0.05
+        should_repeat = True
+        result = None
+
+        while should_repeat:
+            try:
+                result = func(*args, **kwargs)
+                should_repeat = False
+            except DatabaseError:
+                repeat_handler.is_max_number_of_tries_exceeded()
+                time.sleep(wait_time_in_sec)
```

Comment:
I tested it and instant retry requires 20-30 retries, when i test it in 2-3 threads. The sleep time (should be set in settings as well) set as 0.05 second is like 'instant', but requires only 3-5 retries.

Reply:
It will depend on what is inside the critical section.

A good strategy for waiting is to start with something small and double it after each try. For example Kubernetes does it like this.